### PR TITLE
Reduce casting for objects

### PR DESCRIPTION
**Pull request by Gymnasiast**
Using a trick I nicked from the tile elements, I was able to massive reduce the need for static casts on retrieving objects.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
oh lol that was going to be my next PR

**Review by duncanspumpkin — APPROVED** [review]

**Review by tupaschoal — APPROVED** [review]
Nice!

**Comment by Gymnasiast** [comment]
> oh lol that was going to be my next PR

It was your earlier PR that give the inspiration. :D

File: src/openrct2/object/AudioObject.h [hunk 1]
```diff
@@ -23,6 +23,8 @@ class AudioObject final : public Object
     AudioSampleTable _loadedSampleTable;
 
 public:
+    static constexpr ObjectType objectType = ObjectType::Audio;
```
Comment (ZehMatt) [inline src/openrct2/object/AudioObject.h:26]: I think this should become `kObjectType`

**Pull request merged**